lib/CATS/Judge/DirectDatabase.pm

```diff
@@ -81,8 +63,30 @@ sub set_request_state {
     $dbh->commit;
 }
 
+sub was_pinged {
+    my ($self) = @_;
+
+    $self->{was_pinged};
+}
+
```

Comment:
One-liner would be good here, see `is_locked` above.
I even suggest moving this into `Judge::Base` to avoid duplicate in `WebApi` below.